File: apps/cpp-lib/greeter_test.cc
#include "apps/cpp-lib/greeter.h"
#include <gtest/gtest.h>

namespace example {

TEST(GreeterTest, GreetsWithName) {
    Greeter g("Bazel");
    // Greeting embeds build info from //libs/common:version (shared library dep).
    // Use find() to check the prefix without hard-coding the full version string.
    EXPECT_NE(g.Greet().find("Hello from C++, Bazel!"), std::string::npos);
}

TEST(GreeterTest, GreetsWorld) {
    Greeter g("World");
    EXPECT_NE(g.Greet().find("Hello from C++, World!"), std::string::npos);
}

}  // namespace example

File: apps/python-lib/greeter_test.py
import unittest
from greeter import greet

class TestGreet(unittest.TestCase):
    def test_greet(self):
        self.assertEqual(greet("Bazel"), "Hello from Python, Bazel!")

    def test_greet_world(self):
        self.assertEqual(greet("World"), "Hello from Python, World!")

if __name__ == "__main__":
    unittest.main()
